apps/trading/components/swap/swap-form.tsx

```diff
@@ -0,0 +1,386 @@
+import { Link } from 'react-router-dom';
+import {
+  Intent,
+  TradingButton,
+  VegaIcon,
+  VegaIconNames,
+} from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useT } from '../../lib/use-t';
+import {
+  addDecimal,
+  removeDecimal,
+  roundUpToTickSize,
+  toBigNum,
+} from '@vegaprotocol/utils';
+import { OrderTimeInForce, OrderType, Side } from '@vegaprotocol/types';
+import { AssetInput, SwapButton, PriceImpactInput } from './swap-form-elements';
+import BigNumber from 'bignumber.js';
+import { Links } from '../../lib/links';
+import type { Account } from '@vegaprotocol/accounts';
+import type { AssetFieldsFragment } from '@vegaprotocol/assets';
+import {
+  getBaseAsset,
+  getQuoteAsset,
+  type MarketDataFieldsFragment,
+  type MarketFieldsFragment,
+} from '@vegaprotocol/markets';
+import { useVegaTransactionStore } from '@vegaprotocol/web3';
+import { getNotionalSize } from '@vegaprotocol/deal-ticket';
+import { usePrevious } from '@vegaprotocol/react-helpers';
+import { SpotData } from './spot-data';
+
+const getQuoteAssetBalance = (
+  asset?: AssetFieldsFragment,
+  accounts?: Account[] | null
+) => {
+  if (!asset || !accounts) return undefined;
+  const account = accounts.find((a) => a.asset.id === asset.id);
+  return account?.balance;
+};
+
+const derivePrice = (
+  marketData: MarketDataFieldsFragment | null,
+  side: Side,
+  toleranceFactor: number,
+  market: MarketFieldsFragment
+) => {
+  if (!marketData) return;
+  const isBid = side === Side.SIDE_SELL;
+  const price = isBid
+    ? new BigNumber(marketData.bestBidPrice).times(1 - toleranceFactor)
+    : new BigNumber(marketData.bestOfferPrice).times(1 + toleranceFactor);
+  return roundUpToTickSize(price, market.tickSize, isBid);
+};
+
+export const SwapForm = ({
+  marketData,
+  baseAsset,
+  quoteAsset,
+  market,
+  accounts,
+  assets: spotAssets,
+  setBaseAsset,
+  setQuoteAsset,
+}: {
+  market?: MarketFieldsFragment;
+  marketData: MarketDataFieldsFragment | null;
+  assets: AssetFieldsFragment[];
+  baseAsset?: AssetFieldsFragment;
+  quoteAsset?: AssetFieldsFragment;
+  accounts?: Account[] | null;
+  setBaseAsset: (asset?: AssetFieldsFragment) => void;
+  setQuoteAsset: (asset?: AssetFieldsFragment) => void;
+}) => {
+  const t = useT();
+
+  const [quoteAmount, setQuoteAmount] = useState('');
+  const [baseAmount, setBaseAmount] = useState('');
+  const [tolerance, setTolerance] = useState('');
+
+  const { pubKey, isReadOnly } = useVegaWallet();
+  const create = useVegaTransactionStore((state) => state.create);
+
+  const quoteAssetBalance = getQuoteAssetBalance(quoteAsset, accounts);
+  const baseAssetBalance = getQuoteAssetBalance(baseAsset, accounts);
+
+  const side = useSide({ market, baseAsset, quoteAsset });
+
+  const marketPrice = useMarketPrice({ marketData, side });
+  const prevMarketPrice = usePrevious(marketPrice);
+
+  const handleSwapAssets = () => {
```

Comment:
I don't like that it's confusing. BASE and QUOTE are always the same for given asset pair (unless there's a market with flipped assets).

This should only change the SIDE of the order BUY / SELL

Reply:
we have another issue here though that I am about to resolve. 

the size set up is not correct and we need to fix this.

Reply:
changes incoming

Reply:
updated based on your feedback